transformer/compose/v3.go

```diff
@@ -80,6 +81,21 @@ func removeNonExistentEnvFilesV3(path string, parsedComposeFile map[string]inter
 							}
 							vals[envFile] = envfiles
 						}
+					} else {
+						checkForDotEnv = true
+					}
+				} else {
+					checkForDotEnv = true
+				}
+				if checkForDotEnv {
+					vals := val.(map[string]interface{})
+					envFilePath := filepath.Join(composeFileDir, defaultEnvFile)
+					finfo, err := os.Stat(envFilePath)
+					if os.IsNotExist(err) || finfo.IsDir() {
+						logrus.Debugf("Unable to find env config file %s referred in service %s in file %s. Ignoring it.", envFilePath, serviceName, path)
```

Comment:
If we are here, it means it is not referenced in the docker compose file, isn't it?

Reply:
if we hit this condition, it means that docker compose has not listed any env file. So, as a follow-on, we also check if .env is available or not. If available we add it to the env list, if the file does not exist, then we dont add anything.